### DISCUSSION TITLE
wxt 0.19 how to remove "entrypointLoader: 'jiti"

### DISCUSSION THREAD

**Discussion by yebei199** [wxt-dev/wxt#861] [Q&A] [answered]
When I write code like this, it throws an error, even after I remove the polyfill. The tutorial didn't tell me what to do about it.
it's Get back to normal work when i add' entrypointLoader: 'jiti', but NOTE: "jiti" is deprecated and will be removed in the next major version, v0.20.0.

```
import { defineExtensionStorage } from '@webext-core/storage'
import { browser } from 'wxt/browser'

export interface IWordStorage {
  word: string
  definition?: string
  example?: string
  isDeleted: boolean
  queryTimes: number
  deleteTimes: number
}
export interface IALlWordsStorage {
  [key: string]: IWordStorage
}

export interface ExtensionStorageSchema {
  myWords: IALlWordsStorage
}

export const extensionStorage =
  defineExtensionStorage<ExtensionStorageSchema>(
    browser.storage.local
  )
`
`import { defineExtensionMessaging } from '@webext-core/messaging'
import { IWordQuery } from '@/entrypoints/trans.content/script/TransLine.tsx'

interface ProtocolMap {
  trans(data: IWordQuery): string
}

export const { sendMessage, onMessage } =
  defineExtensionMessaging<ProtocolMap>()

```
error:

> 
>  ERROR  This script should only be loaded in a browser extension.                                                                                                                                                                      10:08:46
> 
>   at node_modules\.pnpm\webextension-polyfill@0.10.0\node_modules\webextension-polyfill\dist\browser-polyfill.js:10:9
>   at node_modules\.pnpm\webextension-polyfill@0.10.0\node_modules\webextension-polyfill\dist\browser-polyfill.js:5:5
>   at Object.<anonymous> (node_modules\.pnpm\webextension-polyfill@0.10.0\node_modules\webextension-polyfill\dist\browser-polyfill.js:13:3)
>   at Module._compile (node:internal/modules/cjs/loader:1358:14)
>   at Object.Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
>   at Module.load (node:internal/modules/cjs/loader:1208:32)
>   at Function.Module._load (node:internal/modules/cjs/loader:1024:12)
>   at cjsLoader (node:internal/modules/esm/translators:348:17)
>   at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:297:7)
>   at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
>   at ModuleLoader.import (node:internal/modules/esm/loader:316:24)
>   at ViteNodeRunner.interopedImport (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:421:28)
>   at ViteNodeRunner.directRequest (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:280:24)
>   at ViteNodeRunner.cachedRequest (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:206:14)
>   at ViteNodeRunner.dependencyRequest (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:259:12)
>   at src\wxtMessaging.ts:1:31
>   at ViteNodeRunner.runModule (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:399:5)
>   at ViteNodeRunner.directRequest (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:381:5)
>   at ViteNodeRunner.cachedRequest (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:206:14)
>   at ViteNodeRunner.dependencyRequest (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:259:12)
>   at entrypoints\background.ts:3:31
>   at ViteNodeRunner.runModule (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:399:5)
>   at ViteNodeRunner.directRequest (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:381:5)
>   at ViteNodeRunner.cachedRequest (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:206:14)
>   at ViteNodeRunner.executeFile (/D:/A_sf_py/my-trans/node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.11/node_modules/vite-node/dist/client.mjs:169:12)
>   at Object.importEntrypoint (/D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/core/builders/vite/index.mjs:203:23)
>   at getBackgroundEntrypoint (/D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/core/utils/building/find-entrypoints.mjs:251:27)
>   at /D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/core/utils/building/find-entrypoints.mjs:55:18
>   at async Promise.all (index 0)
>   at findEntrypoints (/D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/core/utils/building/find-entrypoints.mjs:43:23)
>   at internalBuild (/D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/core/utils/building/internal-build.mjs:30:23)
>   at buildAndOpenBrowser (/D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/core/create-server.mjs:77:28)
>   at Object.start (/D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/core/create-server.mjs:42:9)
>   at /D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/cli/commands.mjs:32:5
>   at CAC.<anonymous> (/D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/cli/cli-utils.mjs:16:22)
>   at /D:/A_sf_py/my-trans/node_modules/.pnpm/wxt@0.19.0_@types+node@20.14.11_rollup@4.19.0/node_modules/wxt/dist/cli/index.mjs:11:1

![image](https://github.com/user-attachments/assets/7231d20a-65b5-440d-be1f-76ee06a7fb21)

![image](https://github.com/user-attachments/assets/0b8bb3a3-5010-4cea-8fb8-dc78b74b5458)

**Comment by yebei199** [comment]
it's Get back to normal work when i add'  entrypointLoader: 'jiti', but NOTE: "jiti" is deprecated and will be removed in the next major version, v0.20.0.

**Comment by aklinker1** [comment, marked as answer]
_Edit: See https://wxt.dev/guide/upgrade-guide/wxt#vite-node-entrypoint-loader_

Add `@webext-core/storage` to `vite.ssr.noExternal`.

I'll update the docs with a full example and correct the docs when I get home in about an hour

**Reply by yebei199** [reply]
> Add `@webext-core/storage` to `vite.ssr.noExternal`.
> 
> I'll update the docs with a full example and correct the docs when I get home in about an hour

Thanks a lot, everything is fine now
![image](https://github.com/user-attachments/assets/786f2277-2d5b-4d2a-aa7a-227159e07167)

**Reply by aklinker1** [reply]
Yup, exactly! Here are the docs I forgot to update: https://wxt.dev/guide/go-further/entrypoint-loaders.html#vite-node

And here's the new upgrade guide: https://wxt.dev/guide/upgrade-guide/wxt#vite-node-entrypoint-loader
